pydoctor/templatewriter/pages/sidebar.py

```diff
@@ -125,39 +127,43 @@ def __init__(self, loader: ITemplateLoader, ob: Documentable, documented_ob: Doc
         self._depth = depth
         self._level = level + 1
 
-        self.classList = self._getListOf(Class)
-        self.functionList = self._getListOf(Function)
-        self.variableList = self._getListOf(Attribute)
-        self.subModuleList = self._getListOf(Module)
+        _direct_children = self._children(inherited=False)
 
-        self.inheritedFunctionList = self._getListOf(Function, inherited=True) if isinstance(self.ob, Class) else None
-        self.inheritedVariableList = self._getListOf(Attribute, inherited=True) if isinstance(self.ob, Class) else None
-    
-    def _getListOf(self, type_: Type[Documentable],
-                         inherited: bool = False) -> Optional['ContentList']:
-        children = self._children(inherited=inherited)
-        if children:
-            things = [ child for child in children if isinstance(child, type_) ]
-            return self._getListFrom(things, expand=self._isExpandable(type_))
-        else:
-            return None
+        self.classList = self._getContentList(_direct_children, Class)
+        self.functionList = self._getContentList(_direct_children, Function)
+        self.variableList = self._getContentList(_direct_children, Attribute)
+        self.subModuleList = self._getContentList(_direct_children, Module)
+        
+        self.inheritedFunctionList: Optional[ContentList] = None
+        self.inheritedVariableList: Optional[ContentList] = None
 
+        if isinstance(self.ob, Class):
```

Comment:
Longer-term question: could this be added as a method to `Documentable` so we can just call it on `self.ob` rather than dispatching on type?